brain/compartmentReportView.cpp

```diff
@@ -119,10 +55,30 @@ const CompartmentReportMapping& CompartmentReportView::getMapping() const
     return _impl->mapping;
 }
 
+// using float is necessary to match brion::CompartmentReport
+// frame index calculation
+inline size_t getFrameIndex(float t, float start, float timestep)
+{
+    return std::floor((t - start) / timestep);
+}
+
 std::future<CompartmentReportFrame> CompartmentReportView::load(
-    const float timestamp)
+    double timestamp)
 {
+    const double start = _impl->report->getStartTime();
+    const double end = _impl->report->getEndTime();
+
+    if (timestamp < start || timestamp >= end)
+        throw std::logic_error("Invalid timestamp");
+
+    const double timestep = _impl->report->getTimestep();
+
+    size_t frameIndex = getFrameIndex(timestamp, start, timestep);
+
+    timestamp = start + frameIndex * timestep;
```

Comment:
Since you only use getFrameIndex here, you could move this sentence to it and rename it to _snapTimestamp, _snapToTimeResolution or something similar